src/Features/Core/Portable/Completion/Providers/AbstractObjectInitializerCompletionProvider.cs

```diff
@@ -62,8 +62,9 @@ internal abstract class AbstractObjectInitializerCompletionProvider : CommonComp
 
             foreach (var uninitializedMember in uninitializedMembers)
             {
+                var displayText = uninitializedMember.GetParameters().Length == 0 ? uninitializedMember.ToNameDisplayString() : uninitializedMember.Name;
```

Comment:
I'm not sure that checking the number of parameters is the correct way to filter out such members. It feels like we should look at the kind of member (field or properties are good for this completion, but methods are not).

Reply:
Agreed.  This is patching things up at the wrong level.  We have a mechanism to do this already in normal symbol completion.  But unfortunately, we're not doing the same thing here.  The right fix is to adopt that strategy.  @mikkelbu I can provide more information if my explanation so far wasn't clear.

Reply:
@CyrusNajmabadi I think I need some more information. Originally when I looked into this issue I was also by `AbstractSymbolCompletionProvider` and I looked at `CompletionUtilities.GetDisplayAndInsertionText`, but I failed to discover that `CompletionUtilities` exists both as a C# and a VB class, so I took another approach.

Are you suggesting that I make a similar abstract method in `AbstractObjectINitializerCompletionProvider` and then use `CompletionUtilities.GetDisplayAndInsertionText` in the derived classes in C# and VB? And if so, should I also have something similar to the special handling in `ObjectCreationCompletionProvider.GetDisplayAndInsertionText(...)` and `AbstractObjectCreationCompletionProvider.GetDisplayAndInsertionText(...)`.

Ps. `GetDisplayAndInsertionText` is using `EscapeIdentifier` to create the insertion text.

Reply:
That was my original suggestion for the overall 'best' approach here.

However, i think julien's suggestions is just as reasonable and far easier to accomplish.  So i would just go with that.

Cheers!